dbms/src/Common/UTF8Helpers.cpp

```diff
@@ -0,0 +1,28 @@
+#include <Common/UTF8Helpers.h>
+
+#include <locale>
+#include <codecvt>
+#include <widechar_width.h>
+
+namespace DB
+{
+namespace UTF8
+{
+
+size_t computeWidth(const UInt8 * data, size_t size)
+{
+    std::wstring wstr = std::wstring_convert<std::codecvt_utf8<wchar_t>, wchar_t>{}.from_bytes(reinterpret_cast<const char *>(data), reinterpret_cast<const char *>(data + size));
```

Comment:
I've read that it will throw on invalid UTF-8 sequence:
https://en.cppreference.com/w/cpp/locale/wstring_convert/from_bytes
It's better to use replacement character �
(as your terminal do when invalid sequence is received)